plugin-name/includes/class-plugin-name-deactivator.php

```diff
@@ -22,14 +23,132 @@
 class Plugin_Name_Deactivator {
 
 	/**
-	 * Short Description. (use period)
+	 * The $_REQUEST during plugin activation.
 	 *
-	 * Long Description.
+	 * @since    1.0.0
+	 * @access   private
+	 * @var      array    $request    The $_REQUEST array during plugin activation.
+	 */
+	private static $request = array();
+
+	/**
+	 * The $_REQUEST['plugin'] during plugin activation.
+	 *
+	 * @since    1.0.0
+	 * @access   private
+	 * @var      string    $plugin    The $_REQUEST['plugin'] value during plugin activation.
+	 */
+	private static $plugin  = 'plugin-name/plugin-name.php';
+
+	/**
+	 * The $_REQUEST['action'] during plugin activation.
+	 *
+	 * @since    1.0.0
+	 * @access   private
+	 * @var      array    $action    The $_REQUEST[action] value during plugin activation.
+	 */
+	private static $action  = 'deactivate';
+
+	/**
+	 * Activate the plugin.
+	 *
+	 * Checks if the plugin was (safely) activated.
+	 * Place to add any custom action during plugin activation.
 	 *
 	 * @since    1.0.0
 	 */
 	public static function deactivate() {
 
+		if ( false === self::get_request()
+			|| false === self::validate_request( self::$plugin, self::$action )
+			|| false === self::check_caps()
+			|| ! check_admin_referer( 'deactivate-plugin_' . self::$request['plugin'] )
+		 ) {
+
+			exit;
+
+		}
+
+		/**
+		 * The plugin is now safely activated.
+		 * Perform your activation actions here.
+		 */
+
+	}
+
+	/**
+	 * Get the request.
+	 *
+	 * Gets the $_REQUEST array and checks if necessary keys are set.
+	 * Populates self::request with necessary and sanitized values.
+	 *
+	 * @since    1.0.0
+	 * @return bool|array false or self::$request array.
+	 */
+	private static function get_request() {
+
+		if ( ! empty( $_REQUEST ) ) {
```

Comment:
This is garbage. How could a nonce be verified _before_ we even checked if the request exists? Additionally this warning does not happen on local WPCS instance. If we would check for nonce first, it would tell us "possibly unset global", because `_wpnonce` might not be set, in request.

The only solution to this is to put the entire check (if is set, if exits, if nonce is valid, if action and plugin are set, in _one_ line, and that gets crazy messy.

However the only chance to silence PHPCS. _Weird_ that locally it does not complain. I hope this will do it ....